assets/blocks/restricted-content/index.js

```diff
@@ -0,0 +1,115 @@
+import { __ } from '@wordpress/i18n';
+import { Icon } from '@wordpress/components';
+import edit from './edit';
+import save from './save';
+import unenrolledMetadata from './unenrolled/block';
+import enrolledMetadata from './enrolled/block';
+import { createBlock } from '@wordpress/blocks';
+
+/**
+ * Helper method to create transform objects.
+ *
+ * @param {string} blockName        The block name to transform to.
+ * @param {string} inverseBlockName The name of the block which is the inverse of the one to transform to.
+ */
+const createWrapTransform = ( blockName, inverseBlockName ) => ( {
+	from: [
+		{
+			type: 'block',
+			isMultiBlock: true,
+			blocks: [ '*' ],
+			__experimentalConvert: ( blocks ) => {
+				if ( blocks.length === 1 && blocks[ 0 ].name === blockName ) {
+					return;
+				}
+
+				// The conversion is done by creating a wrapper block and setting the selected blocks as inner blocks.
+				// If however there is at least one block selected which is the inverse of the one that we transform to,
+				// a wrapper block is not created and we just switch the inverse block(s).
+				let hasReversed = false;
+
+				const wrapperInnerBlocks = blocks.map( ( block ) => {
+					let isReversed = false;
+
+					if ( inverseBlockName === block.name ) {
+						hasReversed = true;
+						isReversed = true;
+					}
+
+					return createBlock(
+						isReversed ? blockName : block.name,
+						block.attributes,
+						block.innerBlocks
+					);
+				} );
+
+				if ( hasReversed ) {
+					return wrapperInnerBlocks;
+				}
+
+				const alignments = [ 'wide', 'full' ];
+
+				// Determine the widest setting of all the blocks to be grouped.
+				const widestAlignment = blocks.reduce( ( result, block ) => {
+					const { align } = block.attributes;
+					return alignments.indexOf( align ) >
+						alignments.indexOf( result )
+						? align
+						: result;
+				}, undefined );
+
+				return createBlock(
+					blockName,
+					{
+						align: widestAlignment,
+					},
+					wrapperInnerBlocks
+				);
+			},
+		},
+	],
+} );
+
+export const EnrolledContent = {
+	title: __( 'Enrolled Content', 'sensei-lms' ),
+	description: __(
+		'Content inside this container block will be displayed to enrolled users only.',
+		'sensei-lms'
+	),
+	keywords: [
+		__( 'Enrolled', 'sensei-lms' ),
+		__( 'Content', 'sensei-lms' ),
+		__( 'Locked', 'sensei-lms' ),
+		__( 'Private', 'sensei-lms' ),
+	],
+	icon: () => <Icon icon="lock" />,
+	edit,
+	save,
+	transforms: createWrapTransform(
+		'sensei-lms/enrolled-content',
+		'sensei-lms/unenrolled-content'
+	),
+	...enrolledMetadata,
+};
+
+export const UnenrolledContent = {
+	title: __( 'Unenrolled Content', 'sensei-lms' ),
+	description: __(
+		'Content inside this container block will be displayed to unenrolled users only.',
```

Comment:
I don't think we use `unenrolled` anywhere, something like `..users not enrolled` might sound better.

Reply:
Fixed in 94e3470fb49c41a0f64056b0bd2aa6bcd7859bcc. I don't like `unenrolled` too but couldn't find a better alternative. Hoping for a better recommendation by the design review cc @donnapep